tasmota/tasmota_xsns_sensor/xsns_100_ina3221.ino

```diff
@@ -30,16 +30,41 @@
  * IMPORTANT INFORMATION
  * By default the driver is enabled to support up to 4 INA3221 from hte above addresse
  * If you want to enable less addresses in order to use other I2C chip on those addresses you
- * can define in your user_config_override.h the following:
- * #define INA3221_MAX_COUNT      the max number of INA3221 to support
- * #define INA3221_ADDRESS1       the I2C address of the 1st INA3221
+ *   can define in your user_config_override.h the following:
+ *   #define INA3221_MAX_COUNT      the max number of INA3221 to support
+ *   #define INA3221_ADDRESS1       the I2C address of the 1st INA3221
  * For example to support only 2 INA3221 at addresses 0x41 and 0x42 you can use:
- * #define INA3221_MAX_COUNT      2
- * #define INA3221_ADDRESS1       0x41
- * That would leave 0x40 and 0x42 for other devices
+ *   #define INA3221_MAX_COUNT      2
+ *   #define INA3221_ADDRESS1       0x41
+ *   That would leave 0x40 and 0x43 for other devices
+ * By defining INA3221_CALC_CHARGE_AH INA3221_CALC_ENERGY_WH the driver adds a estimation of 
+ *     energies in Ah and Wh to the output, To reset the energie calculation disable tge 
+ *     according INA3221 chanel by setting the shunt to 0.0 and enable it again 
+ * For example :
+ *     sensor100 1,0.0,0.1,0.2 and sensor100 1,0.1,0.1,0.2 will reset chanel 1
  * Nevertheless, hte driver tries to identifiy if the chip as an address is a IN3221
 \*********************************************************************************************/
 
+// setup of INA3221 config
+
+#ifndef INA3221_AVERAGING_MASK
+#define INA3221_AVERAGING_MASK              INA3221_AVERAGING_512_SAMPLES   
+#endif
+#ifndef INA3221_VBUS_CONV_TIME_MASK
+#define INA3221_VBUS_CONV_TIME_MASK         INA3221_VBUS_CONV_TIME_8MS    
+#endif
+#ifndef INA3221_SHUNT_CONV_TIME_MASK
+#define INA3221_SHUNT_CONV_TIME_MASK        INA3221_SHUNT_CONV_TIME_4MS    
+#endif
+#ifndef INA3221_MODE_MASK
+#define INA3221_MODE_MASK                   INA3221_MODE_SHUNT_AND_BUS_CONTINOUS    
+#endif
+
+#define  D_CHARGE_AH                        D_BATTERY_CHARGE    
+#define  D_ENERGY_WH                        D_AS3935_ENERGY
```

Comment:
Why referring to AS3935 in INA3221 driver

Reply:
necause there was the text already

Reply:
That's confusing
Better to add a proper `D_ENERGY` along with D_VOLTAGE ...